src/jdk.jlink/share/classes/jdk/tools/jlink/internal/plugins/StripJavaDebugAttributesPlugin.java

```diff
@@ -67,13 +107,19 @@ public ResourcePool transform(ResourcePool in, ResourcePoolBuilder out) {
                         var clm = newClassReader(path, resource,
                                 ClassFile.DebugElementsOption.DROP_DEBUG,
                                 ClassFile.LineNumbersOption.DROP_LINE_NUMBERS);
+
+                        MethodTransform mt;
+                        if (isDroppingMethodNames) {
+                            mt = MethodTransform.dropping(me -> me instanceof MethodParametersAttribute)
```

Comment:
I know this is already existing behavior, but removing the attribute doesn't just remove the parameter names but also the parameter flags. This results in the same bug that was fixed by https://bugs.openjdk.org/browse/JDK-8292275 (or for a more compact description, see the CSR https://bugs.openjdk.org/browse/JDK-8292467).

But that should be addressed separately.